src/components/NewCommentForm.tsx

```diff
@@ -1,6 +1,89 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  postid: number;
+  isLoadingComments: boolean;
+  onSubmit: (comment: Comment) => void;
+  commentLoading: (marker: boolean) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  postid,
+  onSubmit,
+  commentLoading,
+  isLoadingComments,
+}) => {
+  const [authorName, setAuthorName] = useState('');
+  const [authorNameError, setAuthorNameError] = useState('');
+
+  const [authorEmail, setAuthorEmail] = useState('');
+  const [authorEmailError, setAuthorEmailError] = useState('');
+
+  const [commentText, setCommentText] = useState('');
+  const [commentTextError, setCommentTextError] = useState('');
+
+  const handleAuthorNameChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setAuthorName(event.target.value);
+    setAuthorNameError('');
+  };
+
+  const handleAuthorEmailChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setAuthorEmail(event.target.value);
+    setAuthorEmailError('');
+  };
+
+  const handleCommentTextChange = (
+    event: React.ChangeEvent<HTMLTextAreaElement>,
+  ) => {
+    setCommentText(event.target.value);
+    setCommentTextError('');
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    commentLoading(true);
+    if (!authorName) {
+      setAuthorNameError('Name is required');
+    }
+
+    if (!authorEmail) {
+      setAuthorEmailError('Email is required');
+    }
+
+    if (!commentText) {
+      setCommentTextError('Enter some text');
+    }
+
+    if (!authorName || !authorEmail || !commentText) {
+      return;
+    }
+
+    onSubmit({
+      id: 0,
+      postId: postid,
+      name: authorName,
+      email: authorEmail,
+      body: commentText,
+    });
+    commentLoading(false);
```

Comment:
The `commentLoading` function is called twice: once at the beginning of `handleSubmit` and once at the end. Consider whether both calls are necessary. If the loading state should only be set during the API call, you might want to move the second call inside the `onSubmit` function or after it completes.